### ISSUE TITLE
Bug: Docker seems to miss the new single-directory feature

### ISSUE THREAD

**Issue by martijn-brant** [abraunegg/onedrive#2407]
### Describe the bug

The driveone/onedrive docker image seems to have a different version for entrypoint.sh and it's missing the part for single directory env variable in Docker. Entrypoint.sh (https://github.com/abraunegg/onedrive/blob/b3829c1ef340c2869a3475d231e42394e4f3bbbc/contrib/docker/entrypoint.sh) needs the part for adding the single-directory parameter. It's present in the Github code but the Docker image driveone/onedrive: latest is missing the following code:

`# Tell client to use sync single dir option
if [ -n "${ONEDRIVE_SINGLE_DIRECTORY:=""}" ]; then
   echo "# We are synchronizing in single-directory mode"
   echo "# Adding --single-directory ARG"
   ARGS=(--single-directory \"${ONEDRIVE_SINGLE_DIRECTORY}\" ${ARGS[@]})
fi`

Also, I have to change that last line to this for it to work

`   ARGS=(--single-directory ${ONEDRIVE_SINGLE_DIRECTORY} ${ARGS[@]})`

### Operating System Details

```shell
Unraid with latest docker and WSL 2 (Linux DESKTOP-OBAEDG2 5.15.90.1-microsoft-standard-WSL2 #1 SMP Fri Jan 27 02:56:13 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux) with docker 24.0.2 installed.

Docker "Image": "sha256:01fab2bb840a49ddde6096793471564079b20b45c7fb85f68fb2cdceff150bae"

REPOSITORY          TAG       IMAGE ID       CREATED        SIZE
driveone/onedrive   latest    01fab2bb840a   4 months ago   213MB
```

### Client Installation Method

From Distribution Package

### OneDrive Account Type

Personal

### What is your OneDrive Application Version

v2.4.23

### What is your OneDrive Application Configuration

```shell
onedrive version                             = v2.4.23
Config path                                  = /root/.config/onedrive
Config file found in config path             = false
Config option 'sync_dir'                     = /root/OneDrive
Config option 'enable_logging'               = false
Config option 'log_dir'                      = /var/log/onedrive/
Config option 'disable_notifications'        = false
Config option 'min_notify_changes'           = 5
Config option 'skip_dir'                     =
Config option 'skip_dir_strict_match'        = false
Config option 'skip_file'                    = ~*|.~*|*.tmp
Config option 'skip_dotfiles'                = false
Config option 'skip_symlinks'                = false
Config option 'monitor_interval'             = 300
Config option 'monitor_log_frequency'        = 6
Config option 'monitor_fullscan_frequency'   = 12
Config option 'read_only_auth_scope'         = false
Config option 'dry_run'                      = false
Config option 'upload_only'                  = false
Config option 'download_only'                = false
Config option 'local_first'                  = false
Config option 'check_nosync'                 = false
Config option 'check_nomount'                = false
Config option 'resync'                       = false
Config option 'resync_auth'                  = false
Config option 'cleanup_local_files'          = false
Config option 'classify_as_big_delete'       = 1000
Config option 'disable_upload_validation'    = false
Config option 'bypass_data_preservation'     = false
Config option 'no_remote_delete'             = false
Config option 'remove_source_files'          = false
Config option 'sync_dir_permissions'         = 700
Config option 'sync_file_permissions'        = 600
Config option 'space_reservation'            = 52428800
Config option 'application_id'               =
Config option 'azure_ad_endpoint'            =
Config option 'azure_tenant_id'              = common
Config option 'user_agent'                   =
Config option 'force_http_11'                = false
Config option 'debug_https'                  = false
Config option 'rate_limit'                   = 0
Config option 'operation_timeout'            = 3600
Config option 'sync_root_files'              = false
Selective sync 'sync_list' configured        = false
Config option 'sync_business_shared_folders' = false
Business Shared Folders configured           = false
Config option 'webhook_enabled'              = false
```

### What is your 'curl' version

```shell
curl --version
bash: curl: command not found
```

### Where is your 'sync_dir' located

Local

### What are all your system 'mount points'

```shell
N/a docker
```

### What are all your local file system partition types

```shell
N/a docker
```

### How do you use 'onedrive'

Personal OneDrive used over multiple OS/devices

### Steps to reproduce the behaviour

Try and run the docker image with the ONEDRIVE_SINGLE_DIRECTORY env variable, it doesn't work for the driveone/onedrive docker image hosted on docker hub.

### Complete Verbose Log Output

```shell
I don't have the log anymore but it completely ignores the ENV variable as it's missing in the entrypoint.sh script used in the image hosted on Docker Hub
```

### Other Log Information or Details

```shell
I think this is an issue of the Docker image being out of date/using wrong script for entrypoint.sh?
```

**Comment by martijn-brant** [comment]
To add to this, I'm seeing that the latest tag images are 5 months old and the edge images seem to be a lot fresher. The edge tagged image does have the following code in entrypoint:

`# Tell client to use sync single dir option
if [ -n "${ONEDRIVE_SINGLE_DIRECTORY:=""}" ]; then
   echo "# We are synchronizing in single-directory mode"
   echo "# Adding --single-directory ARG"
   ARGS=(--single-directory \"${ONEDRIVE_SINGLE_DIRECTORY}\" ${ARGS[@]})
fi`

**Comment by abraunegg** [comment]
@martijn-brant 
Please can you re-read this:
https://github.com/abraunegg/onedrive/blob/master/docs/Docker.md

The 'latest' Docker image follows the latest released code - which is v2.4.23

The 'edge' Docker images follow 'master' - so all the commits and changes post last release.

> Does the latest tag need to be updated with a working version without being on the edge?

No - because it is the latest release.

Will there be a new release? Please read this: 

https://github.com/abraunegg/onedrive/releases

**Comment by abraunegg** [comment]
This issue has been automatically locked since there has not been any recent activity after it was closed. Please open a new issue for related bugs.
